Changes to verbiage and adding small bits to the introductory notebook

KellyBlack opened the pull request.

mandli commented:
Interesting, `scipy.misc.factorial` is deprecated and is instead in `scipy.special.factorial`.  Can you make that quick change in `04_error.ipynb`?  Given that Python 2.x support is being dropped by `numpy` and `scipy` we may want to remove the testing for 2.7 but maybe we can ignore that for now.

KellyBlack commented:
I just changed the way factorials are calculated in the 04_error.ipynb notebook. Rather than use scipy I converted it to use numpy. I had to explicitly cast the result of the factorial function to a float32 in the existing function.

mandli commented:
Thanks!
